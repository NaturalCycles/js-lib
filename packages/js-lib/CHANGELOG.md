See [Github Releases](https://github.com/NaturalCycles/js-lib/releases)
