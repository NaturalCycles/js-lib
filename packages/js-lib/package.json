{
  "name": "@naturalcycles/js-lib",
  "type": "module",
  "version": "15.0.0",
  "scripts": {
    "build": "dev-lib build",
    "test": "dev-lib test",
    "lint": "dev-lib lint",
    "bt": "dev-lib bt",
    "lbt": "dev-lib lbt",
    "test-tz1": "TZ=Europe/Stockholm yarn test local",
    "test-tz2": "TZ=JST-9 yarn test local",
    "test-ny": "TZ=GMT-0500 yarn test localTime"
  },
  "dependencies": {
    "tslib": "^2",
    "zod": "^3"
  },
  "devDependencies": {
    "@naturalcycles/dev-lib": "*",
    "@types/crypto-js": "^4",
    "@types/node": "^22",
    "@types/semver": "^7",
    "crypto-js": "^4",
    "dayjs": "^1"
  },
  "files": [
    "dist",
    "cfg",
    "src",
    "!src/test",
    "!src/**/*.test.*",
    "!src/**/__snapshots__"
  ],
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "sideEffects": false,
  "engines": {
    "node": ">=22.12.0"
  },
  "publishConfig": {
    "provenance": true,
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git@github.com:NaturalCycles/js-libs.git",
    "directory": "packages/js-lib"
  },
  "description": "Standard library for universal (browser + Node.js) javascript",
  "keywords": [
    "js-lib",
    "lodash",
    "p-map",
    "standard library"
  ],
  "author": "Natural Cycles Team",
  "license": "MIT"
}
