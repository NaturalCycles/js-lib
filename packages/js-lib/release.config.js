import sharedConfig from '../../release.config.js'

export default sharedConfig
