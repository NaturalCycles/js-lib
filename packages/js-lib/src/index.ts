export * from './abort.js'
export * from './array/array.util.js'
export * from './array/range.js'
export * from './bot.js'
export * from './browser/adminService.js'
export * from './browser/analytics.util.js'
export * from './browser/i18n/fetchTranslationLoader.js'
export * from './browser/i18n/translation.service.js'
export * from './browser/imageFitter.js'
export * from './browser/script.util.js'
export * from './browser/topbar.js'
export * from './datetime/dateInterval.js'
export * from './datetime/localDate.js'
export * from './datetime/localTime.js'
export * from './datetime/timeInterval.js'
export * from './datetime/wallTime.js'
export * from './decorators/asyncMemo.decorator.js'
export * from './decorators/createPromiseDecorator.js'
export * from './decorators/debounce.decorator.js'
export * from './decorators/debounce.js'
export * from './decorators/decorator.util.js'
export * from './decorators/logMethod.decorator.js'
export * from './decorators/memo.decorator.js'
export * from './decorators/memo.util.js'
export * from './decorators/memoFn.js'
export * from './decorators/memoFnAsync.js'
export * from './decorators/retry.decorator.js'
export * from './decorators/timeout.decorator.js'
export * from './define.js'
export * from './deviceIdService.js'
export * from './enum.util.js'
export * from './env.js'
export * from './env/buildInfo.js'
export * from './error/assert.js'
export * from './error/error.model.js'
export * from './error/error.util.js'
export * from './error/errorMode.js'
export * from './error/try.js'
export * from './error/tryCatch.js'
export * from './form.util.js'
export * from './http/fetcher.js'
export * from './http/fetcher.model.js'
export * from './http/http.model.js'
export * from './is.util.js'
export * from './iter/asyncIterable2.js'
export * from './iter/iterable2.js'
export * from './json-schema/from-data/generateJsonSchemaFromData.js'
export * from './json-schema/jsonSchema.cnst.js'
export * from './json-schema/jsonSchema.model.js'
export * from './json-schema/jsonSchema.util.js'
export * from './json-schema/jsonSchemaBuilder.js'
export * from './log/commonLogger.js'
export * from './math/accumulatingAverage.js'
export * from './math/math.util.js'
export * from './math/sma.js'
export * from './math/stack.util.js'
export * from './nanoid.js'
export * from './number/createDeterministicRandom.js'
export * from './number/number.util.js'
export * from './object/deepEquals.js'
export * from './object/map2.js'
export * from './object/object.util.js'
export * from './object/set2.js'
export * from './object/sortObject.js'
export * from './object/sortObjectDeep.js'
export * from './polyfill.js'
export * from './promise/abortable.js'
export * from './promise/pDefer.js'
export * from './promise/pDelay.js'
export * from './promise/pFilter.js'
export * from './promise/pHang.js'
export * from './promise/pMap.js'
export * from './promise/pProps.js'
export * from './promise/pQueue.js'
export * from './promise/pRetry.js'
export * from './promise/pState.js'
export * from './promise/pTimeout.js'
export * from './semver.js'
export * from './string/case.js'
export * from './string/escape.js'
export * from './string/hash.util.js'
export * from './string/json.util.js'
export * from './string/leven.js'
export * from './string/pupa.js'
export * from './string/readingTime.js'
export * from './string/regex.js'
export * from './string/safeJsonStringify.js'
export * from './string/slugify.js'
export * from './string/string.util.js'
export * from './string/stringify.js'
export * from './string/url.util.js'
export * from './time/time.util.js'
export * from './typeFest.js'
export * from './types.js'
export * from './unit/size.util.js'
export * from './web.js'
