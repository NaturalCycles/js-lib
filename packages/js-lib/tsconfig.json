{
  "extends": "@naturalcycles/dev-lib/cfg/tsconfig.json",
  "compilerOptions": {
    "target": "es2023",
    "lib": ["esnext", "dom", "dom.iterable"],
  },
  "include": ["src"],
  "exclude": ["**/__exclude"],
}
