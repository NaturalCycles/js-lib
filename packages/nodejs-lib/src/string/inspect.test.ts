import { mockAllKindsOfThings } from '@naturalcycles/dev-lib/dist/testing/mockAllKindsOfThings.js'
import { _range } from '@naturalcycles/js-lib'
import type { Assertion } from 'vitest'
import { expect, test } from 'vitest'
import { _inspect } from '../index.js'

test('_inspect', () => {
  expectResults(v => _inspect(v), mockAllKindsOfThings()).toMatchSnapshot()
})

test('_inspect maxLen', () => {
  const obj = _range(1, 1000).join(',')
  expect(_inspect(obj, { maxLen: 10 })).toBe(`1,2,3,4,5,... 4 KB message truncated`)
})

function expectResults(fn: (...args: any[]) => any, values: any[]): Assertion {
  // eslint-disable-next-line vitest/valid-expect
  return expect(new Map(values.map(v => [v, fn(v)])))
}
