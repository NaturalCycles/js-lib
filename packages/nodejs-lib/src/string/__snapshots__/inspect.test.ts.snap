// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`_inspect 1`] = `
Map {
  undefined => "undefined",
  null => "null",
  true => "true",
  false => "false",
  NaN => "NaN",
  Infinity => "Infinity",
  "" => "empty_string",
  " " => "empty_string",
  "ho ho ho" => "ho ho ho",
  "{" => "{",
  "{"a": "a1"}" => "{ a: 'a1' }",
  "[5, null, "s"]" => "[ 5, null, 's' ]",
  [
    5,
    null,
    "s",
  ] => "[ 5, null, 's' ]",
  15 => "15",
  [
    1,
    2,
    3,
  ] => "[ 1, 2, 3 ]",
  {
    "a": "a1",
    "b": "b1",
    "c": {
      "d": 25,
      "e": undefined,
    },
  } => "{ a: 'a1', b: 'b1', c: { d: 25, e: undefined } }",
  {
    "documentation_url": "https://developer.github.com/v3/git/refs/#create-a-reference",
    "message": "Reference already exists",
  } => "{
  message: 'Reference already exists',
  documentation_url: 'https://developer.github.com/v3/git/refs/#create-a-reference'
}",
  [Function] => "[Function: abc]",
  /i am regex, who are you\\?/ => "/i am regex, who are you?/",
  Map {
    "a" => "b",
  } => "{ a: 'b' }",
  Set {
    "a",
    "b",
    "c",
  } => "[ 'a', 'b', 'c' ]",
  [Error: some err msg] => "Error: some err msg",
  [Error: Error with code] => "Error: Error with code
code: auth/forbidden",
  [AppError: app error] => "AppError: app error",
  Symbol(a_am_symbol) => "Symbol(a_am_symbol)",
  1984-06-21T00:00:00.000Z => "1984-06-21T00:00:00.000Z",
}
`;
